homeassistant/components/matter/water_valve.py

```diff
@@ -0,0 +1,224 @@
+"""Matter water valve."""
+
+from __future__ import annotations
+
+from enum import IntEnum
+from math import floor
+from typing import Any
+
+from chip.clusters import Objects as clusters
+
+from homeassistant.components.valve import (
+    ATTR_POSITION,
+    ValveDeviceClass,
+    ValveEntity,
+    ValveEntityDescription,
+    ValveEntityFeature,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import Platform
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import LOGGER
+from .entity import MatterEntity
+from .helpers import get_matter
+from .models import MatterDiscoverySchema
+
+ValveConfigurationAndControlFeature = (
+    clusters.ValveConfigurationAndControlFeature.Bitmaps.Feature
+)
+TimeSyncBitmap = clusters.ValveConfigurationAndControl.Bitmaps.TimeSyncBitmap
+LevelBitmap = clusters.ValveConfigurationAndControl.Bitmaps.LevelBitmap
+
+# The MASK used for extracting bits 0 to 1 of the byte.
+OPERATIONAL_STATUS_MASK = 0b11
+
+# map Matter valve types to HA device class
+"""
+TYPE_MAP = {
+    clusters.ValveConfigurationAndControl.Enums.Type.kWater: ValveDeviceClass.WATER, // TODO
+    #clusters.ValveConfigurationAndControl.Enums.Type.kDrapery: ValveDeviceClass.GAS,
+}
+"""
```

Comment:
If this is dynamic, move it to the "calculate_features" logic. We can also extend the discovery schema logic to move it there